diff --git a/Makefile b/Makefile
index f34a015..39c114d 100644
--- a/Makefile
+++ b/Makefile
@@ -88,6 +88,29 @@ else ifeq ($(platform), classic_armv8_a35)
 	ARCH = arm
 	CFLAGS += -march=armv8-a
 	LDFLAGS += -static-libgcc -static-libstdc++
+
+# TRIMUI SMART PRO
+else ifeq ($(platform), tg5040)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   STRIP = $(CROSS_COMPILE)strip
+   NEED_BPP := 16
+   fpic := -fPIC
+   SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+   CFLAGS += -mtune=cortex-a53 -mcpu=cortex-a53 -march=armv8-a
+   CFLAGS += -Ofast \
+   -flto=4 -fwhole-program -fuse-linker-plugin \
+   -fdata-sections -ffunction-sections -Wl,--gc-sections \
+   -fno-stack-protector -fno-ident -fomit-frame-pointer \
+   -falign-functions=1 -falign-jumps=1 -falign-loops=1 \
+   -fno-unwind-tables -fno-asynchronous-unwind-tables -fno-unroll-loops \
+   -fmerge-all-constants -fno-math-errno
+   CXXFLAGS += $(CFLAGS)
+   HAVE_NEON = 1
+   ARCH = arm64
+
 #######################################
 
 else ifeq ($(platform), osx)
@@ -294,7 +317,7 @@ OBJECTS := $(SOURCES_CXX:.cpp=.o) $(SOURCES_C:.c=.o)
 all: $(TARGET)
 
 ifeq ($(DEBUG),0)
-   FLAGS += -O2 -DNDEBUG
+   FLAGS += -O3 -DNDEBUG
 else
    FLAGS += -O0 -g -DDEBUG
 endif
