diff --git a/Makefile b/Makefile
index f5e3bf2..cd27152 100644
--- a/Makefile
+++ b/Makefile
@@ -331,7 +331,7 @@ else ifeq ($(platform), classic_armv7_a7)
 	-falign-functions=1 -falign-jumps=1 -falign-loops=1 \
 	-fno-unwind-tables -fno-asynchronous-unwind-tables -fno-unroll-loops \
 	-fmerge-all-constants -fno-math-errno \
-	-marm -mtune=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
+	-marm -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
 	ASFLAGS += $(CFLAGS)
 	ARCH = arm
 	CPU_ARCH := arm
@@ -512,10 +512,11 @@ else ifeq ($(platform), miyoo)
 	
 
 else ifeq ($(platform), miyoomini)
-	TARGET := $(TARGET_NAME)_plus_libretro.so
-	CC = /opt/miyoomini-toolchain/usr/bin/arm-linux-gcc
-	CXX = /opt/miyoomini-toolchain/usr/bin/arm-linux-g++
-	AR = /opt/miyoomini-toolchain/usr/bin/arm-linux-ar
+	LDFLAGS += -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard -marm
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
 	fpic := -fPIC
 	SHARED := -shared -Wl,--version-script=link.T -Wl,--no-undefined
 	CFLAGS += -Ofast \
@@ -525,7 +526,7 @@ else ifeq ($(platform), miyoomini)
 	-falign-functions=1 -falign-jumps=1 -falign-loops=1 \
 	-fno-unwind-tables -fno-asynchronous-unwind-tables -fno-unroll-loops \
 	-fmerge-all-constants -fno-math-errno \
-	-marm -mtune=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
+	-marm -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
 	CXXFLAGS = $(CFLAGS) -std=gnu++11
 	CPPFLAGS += $(CFLAGS)
 	ASFLAGS += $(CFLAGS)
